src/components/ShoppingCart.js

```diff
@@ -1,22 +1,36 @@
-import React from "react";
+import React, { useEffect } from "react";
+import { IoIosClose } from "react-icons/io";
 
 const ShoppingCart = ({ cart, setCart, updateCart, isOpen, setIsOpen }) => {
-  function handleDecrementBtn(p) {
-    const newcart = cart.filter((item) => {
-      if (item.id === p.id && item.count > 0) {
-        item.count = item.count - 1;
-      }
-      return item;
-    });
+  useEffect(() => {
+    if (cart.length === 0) {
+      setIsOpen(!isOpen);
+    }
+  }, [cart]);
```

Comment:
Beautiful. 

Instead of `setIsOpen(!isOpen)`, you can just do `setIsOpen(false)`; It works the same way, but it prevents opening the cart by mistake ( or a bug).

Reply:
yes makes sense